[Huynh Thai Duong] iP

aidoxe-123 opened the pull request:
A0200676J

Commit:
Add Gradle support

aidoxe-123 committed:
Level-1

aidoxe-123 committed:
Level-2

aidoxe-123 committed:
Add mark as done feature

aidoxe-123 committed:
Add Todo, Event, Deadline

aidoxe-123 committed:
Trying automated text UI testing

aidoxe-123 committed:
Add IncorrectCommandException and InadequateCommandException

aidoxe-123 committed:
Use ArrayList to store tasks

aidoxe-123 committed:
Add delete command

aidoxe-123 committed:
Done, Delete command: Handle invalid index input

aidoxe-123 committed:
Add new test cases

aidoxe-123 committed:
Duke class: add feature: storing tasks locally using .txt file

The tasks are discarded when the program stops

It doesn't help the user to store the tasks in a long period of time

Let's add a basic form of storage system using .txt file

This is the simplest implementation for this program
to store data locally

Might improve it in future commits

aidoxe-123 committed:
Deadline, Event classes: store date using LocalDate class

The deadlines and events' time are being stored using string.

Unable to use the time in a more meaningful way (e.g finding events/
deadlines that is happening on a specific date).

Store time in LocalDate objects whenever possible.

Make it possible to sort tasks by date or look up tasks occuring in a
specific date.

aidoxe-123 committed:
Add find yyyy-MM-dd command

The user doesn't have a way to search for tasks happening on a specific
date.

Add the find command in the form: "find yyyy-MM-dd", which prints out
all tasks that happening on that date

This is a simple command that follows the style of this robot

aidoxe-123 committed:
Merge branch 'branch-Level-7'

aidoxe-123 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

aidoxe-123 committed:
Support multiple date format

aidoxe-123 committed:
Add Storage class

Duke.main is responsible for reading and extracting file

The file is too long

Create a Storage class to deal with this problem

aidoxe-123 committed:
Add Ui class

aidoxe-123 committed:
Rename IncorrectCommandException to DukeException

aidoxe-123 committed:
Add Command classes, finish implementing Ui, Parser, TaskList

aidoxe-123 committed:
Task class: delete Command enums

aidoxe-123 committed:
Organize classes into package

Classes are stored in the same folder.

Kind of look like a mess.

Make it easier to keep track of the project.

aidoxe-123 committed:
Add testing classes for Ui, ToDo and Deadline

aidoxe-123 committed:
Change all '\n' character to System.lineSeperator()

The program is having multiple types of line break

Change '\n' character to System.lineSeperator()

Can run on different systems, easier to write testing classes

aidoxe-123 committed:
Release the first .jar file

aidoxe-123 committed:
Add javadoc for classes in duke.command package

aidoxe-123 committed:
Add javadoc for classes in duke.exception package

aidoxe-123 committed:
Add javadoc for classes in duke.parser package

aidoxe-123 committed:
Add javadoc to classes in duke.task package

aidoxe-123 committed:
Add javadoc to Duke.java, Storage.java and Ui.java

aidoxe-123 committed:
Change files to comply the java coding standard

aidoxe-123 committed:
Add find [keyword] command

aidoxe-123 committed:
Merge branch 'branch-A-JavaDoc'

aidoxe-123 committed:
Merge branch 'branch-A-CodingStandard'

aidoxe-123 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/FindCommand.java

aidoxe-123 committed:
Change all the methods in Ui class from static to non-static

The methods are static

They don't need to be static

aidoxe-123 committed:
Update runtest.bat accordingly

Runtest.bat is only up-to-date with Level-6 tag

Unable to catch up with the new behaviors

Make it up-to-date with the codes in this commit

aidoxe-123 committed:
Update runtest.sh file to catch up with the nearest commit

aidoxe-123 committed:
Allow tasks to be saved to the storage after every command

Task are only saved when bye is pressed

Users may lose their working progress when the program
stops unexpectedly

Save the tasks into the storage after every command

Storage is kept up-to-date

Kind of inefficient, might improve this in future commits

aidoxe-123 committed:
Update data/duke.txt content

aidoxe-123 committed:
Merge branch 'add-gradle-support'

aidoxe-123 committed:
Finish setting up gradle to be able to run program and junit test

aidoxe-123 committed:
Implement checkstyle to check the style of java codes

aidoxe-123 committed:
Merge branch 'branch-A-CheckStyle'

aidoxe-123 committed:
Create GUI interface using Javafx

aidoxe-123 committed:
Use fxml to encode the UI

aidoxe-123 committed:
Make the GUI process commands instead of repeating them

aidoxe-123 committed:
Merge branch 'branch-Level-10'

aidoxe-123 committed:
Fix parsing condition in Parser.java

aidoxe-123 committed:
Use varargs to handle the find command and InadequateCommandException

aidoxe-123 committed:
Merge branch 'branch-A-Varargs'

aidoxe-123 committed:
Fix styling error that violates duke checkstyle

aidoxe-123 committed:
Change font style to Courier New

ngzhenteng commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+            }
+
+            try {
+                Command c = Parser.parse(fullCommand);
```
Maybe a more descriptive name for the variable c such as commandObj could be used?

ngzhenteng commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+        } else if (splitted.length == 1 && splitted[0].equals(ExitCommand.COMMAND)) {
+            isExiting = true;
+        }
+        if (isRunning) {
```
I like how your code checks and maintains the state of the Duke application. This can help prevent the program from behaving in unexpected ways.

ngzhenteng commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,76 @@
+     * @param cmpDate A date that is being queried
+     * @return True if this task is due on that day, false otherwise
+     */
+    public boolean isDueOn(LocalDate cmpDate) {
```
I like the addition of this feature, it could come in handy for the user.

ngzhenteng commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    /**
+     * Executes the command.
```
Maybe it would be better to give a short explanation on how the command is executed?

ngzhenteng commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,35 @@
+import duke.ui.Ui;
+
+/**
+ * Is responsible for handling commands
```
Perhaps more information could be given to explain that Command is a super class for the various types of Command objects, and that the role of a Command object is to create the correct type of Task and pass it to Ui for the correct string output to be printed.

ngzhenteng commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,35 @@
+/**
+ * Is responsible for handling commands
+ */
+public class Command {
```
I like how you have used a class to encompass all types of commands as this is neater and more scalable.

aidoxe-123 replied:
I have updated my code base on your reviews, thank you!

ngzhenteng reviewed (commented):
It is impressive how much thought you have put into the scalability of your code and the organisation of classes. Your code also complies with java coding standards, great job on that. Some ways to improve would be to give more descriptive Javadoc comments so that the reader can understand the way your classes interact with each other since there are many classes in your design. Awesome job, all the best!

zhehaokoh commented on data/duke.txt:
```diff
@@ -0,0 +1,4 @@
+D ND test4 /time 14/02/2020
+E D test /time 1
+T ND finish this                 
+T ND test     
```
I believe that your code has taken care of the situation where the description of the task is a short phrase such as "finish this" indicated in your example, but maybe you can consider using things like | to separate components as demo by the website rather than using a white space. I personally think that will be clearer.

zhehaokoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                isExit = c.isExit();
+                storage.write(tasks); // temporary until having better implementation
```
I like that you add a little comment besides code that might need extra work in the future. But should the comment be placed on top of the code instead of the right side?

aidoxe-123 replied:
I feel that writing tasks into the .txt file for every command is kind of inefficient, so I put the comment on the right of that line to remind me to fix it later. That comment is for line 77 only, not the whole code block.

zhehaokoh commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,41 @@
+ * Is responsible for handling commands starting with <code>deadline</code>.
+ */
+public class DeadlineCommand extends TaskCreationCommand {
+    public static final String COMMAND = "deadline";
+    public static final String TIME_SPECIFIER = "/by";
```
I like your approach that declares keywords as constants upfront.

zhehaokoh commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,53 @@
+
+            } else {
+                filteredTasks = tasks.filter((task) ->
+                        task.toString().toLowerCase().contains(keyword.toLowerCase()));
```
I like your use of Optional and Lambda here. I also like the way you specify a DateParser that is responsible for converting String to a LocalDate.

zhehaokoh commented on src/main/java/duke/command/ToDoCommand.java:
```diff
@@ -0,0 +1,37 @@
+import duke.ui.Ui;
+
+/**
+ * Is responsible for handing commands starting with <code>todo</code>.
```
Maybe it should start with a verb?

aidoxe-123 replied:
I think "is" is a verb

aidoxe-123 replied:
I have updated my code based on your reviews. Thank you for spending time looking through this!

zhehaokoh reviewed (commented):
I think you have fixed all format issues based on CheckStyle, so there is nothing I can comment on coding style violation.

aidoxe-123 committed:
Use css to improve styling of the GUI

aidoxe-123 committed:
Continue improving styling with css

aidoxe-123 committed:
Update Duke.java based on the peer reviews

aidoxe-123 committed:
Allow storage to be updated on the run

aidoxe-123 committed:
Create a new .jar file

aidoxe-123 committed:
duke.command, duke.task, duke.storage package: Add assertions

aidoxe-123 committed:
Improve code quality by extracting methods and changing variable names

aidoxe-123 committed:
duke.task.TaskList class: use stream for the filter method

aidoxe-123 committed:
Merge pull request #1 from aidoxe-123/branch-A-Assertions

duke.command, duke.task, duke.storage package: Add assertions

aidoxe-123 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/Duke.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/TaskList.java

aidoxe-123 committed:
Merge pull request #2 from aidoxe-123/branch-A-CodeQuality

Improve code quality by extracting methods and changing variable names

aidoxe-123 committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/task/TaskList.java

aidoxe-123 committed:
Merge pull request #3 from aidoxe-123/branch-A-Streams

duke.task.TaskList class: use stream for the filter method

aidoxe-123 committed:
Create gradle.yml

aidoxe-123 committed:
Update test files and gradle.yml file

aidoxe-123 committed:
Update gradle.yml

aidoxe-123 committed:
Update gradle.yml

aidoxe-123 committed:
Merge branch 'master' of https://github.com/aidoxe-123/ip

# Conflicts:
#	.github/workflows/gradle.yml

aidoxe-123 committed:
Fix checkstyle errors in the commited files

aidoxe-123 committed:
Fix checkstyle errors

aidoxe-123 committed:
Fix checkstyle problem in DeadlineTest.java

aidoxe-123 committed:
Move files into packages

aidoxe-123 committed:
Allow tasks to be save only when the program exits

aidoxe-123 committed:
Allow the program to undo the previous change to the task list

aidoxe-123 committed:
Merge branch 'branch-C-Undo'

aidoxe-123 committed:
Fix parser bug with keywords having leading and trailing spaces

aidoxe-123 committed:
Delete unused import in Ui.java

aidoxe-123 committed:
Parser class: fix parsing problem with commands having leading/trailing
spaces

aidoxe-123 committed:
Stop tracking duke.txt

aidoxe-123 committed:
Add command: clone [source index] [destination index (optional)]

aidoxe-123 committed:
Add update [index] /description newDescription command

aidoxe-123 committed:
Event class: Fix spelling error (isOccuringOn -> isOccurringOn)

aidoxe-123 committed:
Add update [index] /time [time string] command for Deadline and Event

aidoxe-123 committed:
Merge branch 'branch-C-Update'

aidoxe-123 committed:
Use color to differentiate between normal messages and error messages

aidoxe-123 committed:
Add screen shot of the Ui to the docs folder

aidoxe-123 committed:
Change the message when nothing is found in the find command

aidoxe-123 committed:
Prevent throwing IndexOutOfBoundException when clone/done and then undo

To reproduce the problem:
clone 1 <any index that is larger than [tasks.size() + 1]>
undo
=> IndexOutOfBoundException is thrown
done <any index larger than tasks.size()>
undo
=> IndexOutOfBoundException is thrown

aidoxe-123 committed:
Fix undo command error after update command

aidoxe-123 committed:
Only allow time of an event task to be parsed if the end time happens
from the start time onwards

aidoxe-123 committed:
Add UserGuide

aidoxe-123 committed:
Set theme jekyll-theme-cayman

aidoxe-123 committed:
Add appendix -> pasring date in user guide

aidoxe-123 committed:
Add close command to close the GUI

aidoxe-123 committed:
Add close command into the User Guide

aidoxe-123 committed:
Improve code quality:change multiple if statements to if else statements
